great_expectations/expectations/metrics/column_aggregate_metrics/column_distinct_values.py

```diff
@@ -42,49 +54,67 @@ def _spark(
         metric_value_kwargs: Dict,
         metrics: Dict[str, Any],
         runtime_configuration: Dict,
-    ):
-        observed_value_counts = metrics["column.value_counts"]
-        return set(observed_value_counts.index)
-
-    @classmethod
-    def _get_evaluation_dependencies(
-        cls,
-        metric: MetricConfiguration,
-        configuration: Optional[ExpectationConfiguration] = None,
-        execution_engine: Optional[ExecutionEngine] = None,
-        runtime_configuration: Optional[Dict] = None,
-    ):
-        """Returns a dictionary of given metric names and their corresponding configuration,
-        specifying the metric types and their respective domains"""
-        dependencies: dict = super()._get_evaluation_dependencies(
-            metric=metric,
-            configuration=configuration,
-            execution_engine=execution_engine,
-            runtime_configuration=runtime_configuration,
+    ) -> Set[Any]:
+        df, _, accessor_domain_kwargs = execution_engine.get_compute_domain(
+            metric_domain_kwargs, MetricDomainTypes.COLUMN
+        )
+        column = accessor_domain_kwargs["column"]
+
+        distinct_values = (
+            df.select(column)
+            .distinct()
+            .where(F.col(column).isNotNull())
+            .rdd.flatMap(lambda x: x)
+            .collect()
         )
 
-        if isinstance(
-            execution_engine, (SqlAlchemyExecutionEngine, SparkDFExecutionEngine)
-        ):
-            dependencies["column.value_counts"] = MetricConfiguration(
-                metric_name="column.value_counts",
-                metric_domain_kwargs=metric.metric_domain_kwargs,
-                metric_value_kwargs={
-                    "sort": "value",
-                    "collate": None,
-                },
-            )
-
-        return dependencies
+        return set(distinct_values)
```

Comment:
Is this used as a dependency for some other metric? I'm asking because I often hear that we want to keep data out of gx. That is, I understand we would want distinct counts but what is the use case for the distinct values?

Reply:
It is now a dependency for `column.distinct_values.count`. `column.distinct_values` will be used in Expectations that currently leverage `column.value_counts` where the counts aren't needed. The reason why we can't return a `column_aggregate_partial` here is that there is no PySpark function for `distinct` that can be used on a PySpark `column` object. Instead we have to build the entire query and get the values in the metric implementation.